File: src/core/ask_opus5.py
import requests
import json
import os

api_key = "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"

def call_opus5(prompt):
    headers = {
        "Authorization": f"Bearer {api_key}",
        "Content-Type": "application/json"
    }
    
    data = {
        "model": "anthropic/claude-opus-5",
        "messages": [{"role": "user", "content": prompt}],
        "stream": False
    }
    
    response = requests.post("https://openrouter.ai/api/v1/chat/completions", headers=headers, json=data)
    print("Status:", response.status_code)
    
    if response.status_code == 200:
        result = response.json()
        print(result['choices'][0]['message']['content'])
    else:
        print("Error:", response.text)

if __name__ == "__main__":
    prompt = "Hello Opus 5, are you there?"
    call_opus5(prompt)

File: src/05_all_other_files/ask_opus5.js
import { OpenRouter } from "@openrouter/sdk";
import fs from 'fs';

const openrouter = new OpenRouter({
  apiKey: "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"
});

async function main() {
  const prompt = fs.readFileSync('opus5_prompt.txt', 'utf8');

  try {
    const stream = await openrouter.chat.send({
      chatRequest: {
        model: "anthropic/claude-opus-5", // user specifically requested this
        messages: [
          {
            role: "user",
            content: prompt
          }
        ],
        stream: true
      }
    });

    let response = "";
    for await (const chunk of stream) {
      const content = chunk.choices[0]?.delta?.content;
      if (content) {
        response += content;
        process.stdout.write(content);
      }

      if (chunk.usage) {
        console.log("\nReasoning tokens:", chunk.usage.completionTokensDetails?.reasoningTokens);
      }
    }
  } catch (error) {
    console.error("Error calling OpenRouter:", error.message);
  }
}

main();
